app/src/main/java/org/mifos/mobile/ui/fragments/BeneficiaryApplicationFragment.kt

```diff
@@ -157,52 +127,51 @@ class BeneficiaryApplicationFragment : BaseFragment(), BeneficiaryApplicationVie
+            binding.accountTypeFieldParent.isEnabled = false
+            binding.tilAccountNumber.editText?.setText(beneficiary?.accountNumber)
+            binding.tilAccountNumber.isEnabled = false
+            binding.tilOfficeName.editText?.setText(beneficiary?.officeName)
+            binding.tilOfficeName.isEnabled = false
+            binding.tilBeneficiaryName.editText?.setText(beneficiary?.name)
+            binding.tilTransferLimit.editText?.setText(beneficiary?.transferLimit.toString())
         } else if (beneficiaryState == BeneficiaryState.CREATE_QR) {
-            accountTypeField?.setText(beneficiary?.accountType?.value!!, false)
-            tilAccountNumber?.editText?.setText(beneficiary?.accountNumber)
-            tilOfficeName?.editText?.setText(beneficiary?.officeName)
+            binding.accountTypeField.setText(beneficiary?.accountType?.value!!, false)
+            binding.tilAccountNumber.editText?.setText(beneficiary?.accountNumber)
+            binding.tilOfficeName.editText?.setText(beneficiary?.officeName)
         }
     }
 
     /**
      * Used for submitting a new or updating beneficiary application
      */
-    @OnClick(R.id.btn_beneficiary_submit)
-    fun submitBeneficiary() {
-        tilAccountNumber?.isErrorEnabled = false
-        tilOfficeName?.isErrorEnabled = false
-        tilTransferLimit?.isErrorEnabled = false
-        tilBeneficiaryName?.isErrorEnabled = false
+    private fun submitBeneficiary() {
+        binding.tilAccountNumber.isErrorEnabled = false
+        binding.tilOfficeName.isErrorEnabled = false
+        binding.tilTransferLimit.isErrorEnabled = false
+        binding.tilBeneficiaryName.isErrorEnabled = false
         if (accountTypeId == -1) {
-            Toaster.show(rootView, getString(R.string.choose_account_type))
+            Toaster.show(binding.root, getString(R.string.choose_account_type))
             return
-        } else if (tilAccountNumber?.editText?.text.toString().trim { it <= ' ' } == "") {
-            tilAccountNumber?.error = getString(R.string.enter_account_number)
+        } else if (binding.tilAccountNumber.editText?.text.toString().trim { it <= ' ' } == "") {
+            binding.tilAccountNumber.error = getString(R.string.enter_account_number)
             return
-        } else if (tilOfficeName?.editText?.text.toString().trim { it <= ' ' } == "") {
-            tilOfficeName?.error = getString(R.string.enter_office_name)
+        } else if (binding.tilOfficeName.editText?.text.toString().trim { it <= ' ' } == "") {
+            binding.tilOfficeName.error = getString(R.string.enter_office_name)
             return
-        } else if (tilTransferLimit?.editText?.text.toString() == "") {
-            tilTransferLimit?.error = getString(R.string.enter_transfer_limit)
+        } else if (binding.tilTransferLimit.editText?.text.toString() == "") {
+            binding.tilTransferLimit.error = getString(R.string.enter_transfer_limit)
             return
```

Comment:
You might try using the `with`in  this function as well...

Could we eliminate this `if-else` madness here, and replace them with `when`? I know it would be out of the scope of this PR but it would be great if you could do that... :)

Reply:
Sure @jawidMuhammadi , I'll do this

Reply:
I have made the necessary changes @jawidMuhammadi !!